backend/database/chat.py

```diff
@@ -82,3 +83,27 @@ def get_messages(uid: str, limit: int = 20, offset: int = 0, include_memories: b
         ]
 
     return messages
+    
+
+async def batch_delete_messages(parent_doc_ref, batch_size=450):
+    # batch size is 450 because firebase can perform upto 500 operations in a batch
+    messages_ref = parent_doc_ref.collection('messages')
+    while True:
+        docs = messages_ref.limit(batch_size).stream()
+        docs_list = list(docs)
+        if not docs_list:
+            break
+        batch = db.batch()
+        for doc in docs_list:
+            batch.delete(doc.reference)
+        batch.commit()
+
+
+async def clear_chat( uid: str):
+    try:
+        user_ref = db.collection('users').document(uid)
```

Comment:
don't use db in a router

Reply:
You don't .delete, check the other delete in the code, is setting a flag `{deleted: True}`

Reply:
> don't use db in a router

Yes it's in database directory only

Reply:
> You don't .delete, check the other delete in the code, is setting a flag `{deleted: True}`

Got it